src/blueteam/Archon.java

```diff
@@ -2,34 +2,40 @@
 
 import battlecode.common.Direction;
 import battlecode.common.GameActionException;
-import battlecode.common.MapLocation;
 import battlecode.common.RobotController;
+import battlecode.common.RobotInfo;
 
 public class Archon extends Robot {
 
+	private final int desiredNumberOfGardeners = 10;
+
 	Archon(RobotController rc) {
 		super(rc);
 	}
 
 	@Override
 	void step() throws GameActionException {
-		// Generate a random direction
-		Direction dir = randomDirection();
+		// move away from enemy
+		RobotInfo[] robots = rc.senseNearbyRobots(-1, enemy);
+		if (robots.length > 0) {
+			Direction toEnemy = rc.getLocation().directionTo(robots[0].getLocation());
+			tryMove(toEnemy.opposite());
+		} else {
+			tryMove(randomDirection());
+		}
 
-		// Randomly attempt to build a gardener in this direction
-		if (rc.canHireGardener(dir) && Math.random() < .01) {
+		// randomly attempt to build a gardener if we need more
+		Direction dir = randomDirection();
+		int numOfGardeners = rc.readBroadcast(0);
+		if (numOfGardeners < desiredNumberOfGardeners && rc.canHireGardener(dir) && Math.random() < 0.5) {
```

Comment:
I'm using this to avoid race condition when there are more archons. But we will for sure want to improve this in the future.